src/game.cpp

```diff
@@ -473,12 +473,11 @@ Player* Game::getPlayerByGUID(const uint32_t& guid)
 		return nullptr;
 	}
 
-	for (const auto& it : players) {
-		if (guid == it.second->getGUID()) {
-			return it.second;
-		}
+	auto it = mappedPlayerGuids.find(guid);
```

Comment:
What if `Game::getPlayerByNameWildcard` had not been called before. It would not find a player, wouldn't it?

Reply:
It's not related, in 'Game::addPlayer(Player* player)' is inserted player to 'mappedPlayerGuids' map, exactly same way as 'getPlayerByName' or 'getCreatureByName' etc.

Reply:
Yes, I have no problem with that, as mappedPlayersName is also added.

Reply:
this is an extra cache for improve seach by GUID as the title says. Now `Player(guid)` is fast as `Player(name)` or `Player(cid)`.